src/pytest_bdd/parser.py

```diff
@@ -453,9 +482,50 @@ def parse(self) -> Feature:
 
         for child in feature_data.children:
             if child.background:
-                feature.background = self.parse_background(child.background, feature)
+                feature.background = self.parse_background(child.background)
+            elif child.rule:
+                self._parse_and_add_rule(child.rule, feature)
             elif child.scenario:
-                scenario = self.parse_scenario(child.scenario, feature)
-                feature.scenarios[scenario.name] = scenario
+                self._parse_and_add_scenario(child.scenario, feature)
 
         return feature
+
+    def _parse_and_add_rule(self, rule_data: GherkinRule, feature: Feature) -> None:
+        """Parse a rule, including its background and scenarios, and add to the feature."""
+        rule = Rule(
+            keyword=rule_data.keyword,
+            name=rule_data.name,
+            description=rule_data.description,
+            tags=self.get_tag_names(rule_data.tags),
+        )
+
+        # Add background if present within the rule
+        background = self._extract_rule_background(rule_data)
+        if background:
+            rule.add_background(background)
+
+        # Parse and add scenarios under the rule
+        for rule_scenario in self._extract_rule_scenarios(rule_data):
+            scenario = self.parse_scenario(rule_scenario, feature, rule)
+            feature.scenarios[scenario.name] = scenario
+
+    def _extract_rule_background(self, rule_data: GherkinRule) -> Background | None:
+        """Extract the first background from rule children if it exists."""
+        for child in rule_data.children:
+            if child.background:
+                return self.parse_background(child.background)
+        return None
+
+    @staticmethod
+    def _extract_rule_scenarios(rule_data: GherkinRule) -> Generator[GherkinScenario]:
+        """Yield each scenario under a rule."""
+        for child in rule_data.children:
+            if child.scenario:
+                yield child.scenario
```

Comment:
Shouldn't we `yield self._parse_and_add_scenario(child.scenario)` here instead?

Otherwise it breaks the convention where these `_extract_xxx(...)` methods return an object from the `parser.py` module (instead of a "raw" object from the gherkin_parser.py module)

Reply:
Good point!  Fixed :)